protos/idl/service.proto

```diff
@@ -57,6 +58,16 @@ message AddTagResponse {
 
 }
 
+message ListArtifactsRequest {
+    DatasetID dataset = 1;
+    FilterExpression filter = 2;
+    PaginationOptions pagination = 3;
+}
+
+message ListArtifactsResponse {
+    repeated Artifact artifacts = 1;
```

Comment:
next token?

Reply:
Yes, actually on top of token, it would be helpful to pass back `PaginationOptions` because that will give the caller the context to which to make the next request in terms of `sort_order`, `limit` etc. What do you think of that?

Reply:
I just spoke to Randy about this and he doesn't think from a UI client POV it is that helpful to pass back sort_order and limit as well. But he has seen APIs where the token represents everything needed to get the next page (it includes the filters, sort, limit). I think if we wanted this later on we can add it. I will just pass token back here.